error.c

```diff
@@ -1,14 +1,47 @@
 #include "error.h"
 
+static char **err_msg_ring_buf = NULL;
+static uint8_t err_msg_buf_size = 0;
+static uint8_t err_msg_buf_write = 0;
+static uint8_t err_msg_buf_read = 0;
+
 void report_error(uint8_t board_id,
                   enum BOARD_STATUS error_type,
                   uint8_t byte4, uint8_t byte5,
                   uint8_t byte6, uint8_t byte7)
 {
-    //do nothing
+    if(err_msg_ring_buf == NULL) {
+        err_msg_ring_buf = malloc(
```

Comment:
In real time or embedded systems, use of dynamic allocation is highly discouraged. What we do with all of our other queues is to set up a static array of bytes (usually 100-200 of them), and then use that. If the array gets full, we just drop the message. See https://github.com/waterloo-rocketry/canlib/blob/master/util/can_rcv_buffer.c for an example.

I recommend that you use the safe_ring_buffer module from canlib (you may have to go into your canlib directory and pull from master to get it), but if you want to use your own data structure, you shouldn't use malloc